### PR TITLE
[Dynamo, ONNX] Replace `onnxrt` backend with new backend from ONNXRuntime team 

### PR DESCRIPTION
**Pull request by wschin**
In https://github.com/pytorch/pytorch/pull/106589, a new ONNXRuntime-based Dynamo backend is introduced. As mentioned in that PR, we hope to replace legacy `onnxrt` with that new backend. This PR remove legacy `onnxrt` and register the new backend under the same name.

cc @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @aakhundov

### CODE REVIEW

File: test/onnx/dynamo/test_dynamo_with_onnxruntime_backend.py [hunk 1]
```diff
@@ -64,6 +64,33 @@ def _test_model_numerically(
...
+            example_args_collection,
+        )
+
+        # Test default backend.
+        torch._dynamo.reset()
+        self._test_model_numerically(
+            model,
+            "onnxrt",
+            example_args_collection,
+        )
```
Comment (thiagocrepaldi) [inline test/onnx/dynamo/test_dynamo_with_onnxruntime_backend.py:92]: why not different tests which would call `torch._dynamo.reset()` as part of the `setUp` and `tearDown`?

File: torch/_dynamo/backends/onnxrt.py [hunk 2]
```diff
@@ -1,131 +1,6 @@
-import importlib
-import logging
-import os
-import tempfile
-
-import torch
-from .common import device_from_inputs, fake_tensor_unsupported
+from torch.onnx._internal.onnxruntime import has_onnxruntime, make_aot_ort
```
Comment (thiagocrepaldi) [inline torch/_dynamo/backends/onnxrt.py:1]: should this file have any docstring to list maintainers, etc?

File: test/onnx/dynamo/test_dynamo_with_onnxruntime_backend.py (hunk 1 above)
Comment (wschin) [reply test/onnx/dynamo/test_dynamo_with_onnxruntime_backend.py:92]: I will do
```
    @parameterized.expand(
        [
            (True,),
            (False,),
        ]
    )
    def test_elementwise_function_single_output(self, test_local_backend: bool):
```

and use `test_local_function` to decide which backend to use.

File: torch/_dynamo/backends/onnxrt.py (hunk 2 above)
Comment (wschin) [reply torch/_dynamo/backends/onnxrt.py:1]: Sure. Will do
```
# This backend is maintained by ONNX team. To direct issues
# to the right people, please tag related GitHub issues with `onnx`.
#
# Maintainers' Github IDs: wschin, thiagocrepaldi, BowenBao, abock
from torch.onnx._internal.onnxruntime import has_onnxruntime, make_aot_ort
from .registry import register_backend

```

**Comment by thiagocrepaldi** [comment]
@wschin Will https://github.com/pytorch/pytorch/pull/105826/ be deprecated with the new backend replacement?

File: torch/_dynamo/backends/onnxrt.py
Comment (BowenBao) [inline torch/_dynamo/backends/onnxrt.py:1]: add this file to `.github/merge_rules.yaml` under `ONNX exporter`, such that onnx team can approve and merge PRs on this file going forward.

File: torch/_dynamo/backends/onnxrt.py [hunk 3]
```diff
@@ -1,131 +1,10 @@
...
-import logging
-import os
-import tempfile
-
-import torch
-from .common import device_from_inputs, fake_tensor_unsupported
+# This backend is maintained by ONNX team. To direct issues
+# to the right people, please tag related GitHub issues with `onnx`.
+#
+# Maintainers' Github IDs: wschin, thiagocrepaldi, BowenBao, abock
```
Comment (BowenBao) [inline torch/_dynamo/backends/onnxrt.py:4]: Consider adding this file to CODEOWNERS with the github handles in this line.

PRs modifying this file auto adds code owners as reviewers.

File: torch/_dynamo/backends/onnxrt.py [hunk 4]
```diff
@@ -1,131 +1,10 @@
...
-                value.size(),
-                value.data_ptr(),
-            )
-        session.run_with_iobinding(binding)
-        if device_type == "cpu":
-            binding.copy_outputs_to_cpu()
-        return outputs
-
-    return _call
+if has_onnxruntime():
```
Comment (BowenBao) [inline torch/_dynamo/backends/onnxrt.py:8]: What happens when user call `torch.compile(..., backend="onnxrt") when `has_onnxruntime()` returns False?

**Review by thiagocrepaldi — APPROVED** [review]

File: torch/_dynamo/backends/onnxrt.py (hunk 3 above)
Comment (wschin) [reply torch/_dynamo/backends/onnxrt.py:4]: Ok. Also added myself to onnx files.

File: torch/_dynamo/backends/onnxrt.py (hunk 4 above)
Comment (wschin) [reply torch/_dynamo/backends/onnxrt.py:8]: Likely accessing a non-existing element in a dictionary and then throwing.

File: torch/_dynamo/backends/onnxrt.py
Comment (wschin) [reply torch/_dynamo/backends/onnxrt.py:1]: OK.

File: torch/_dynamo/backends/onnxrt.py (hunk 4 above)
Comment (BowenBao) [reply torch/_dynamo/backends/onnxrt.py:8]: Is it possible to display the error message such that user know it is due to missing dependency?

Comment (wschin) [reply torch/_dynamo/backends/onnxrt.py:8]: It's not `onnxrt`'s job. We need to check dynamo's look up process and modify if necessary.

Comment (wschin) [reply torch/_dynamo/backends/onnxrt.py:8]: Let me take it back. Maybe create an empty runtime and just raise?

**Review by BowenBao — APPROVED** [review]
LGTM w/ suggestions on dependency check.

cc @msaroufim @jansel for approval.

File: torch/_dynamo/backends/onnxrt.py (hunk 4 above)
Comment (wschin) [reply torch/_dynamo/backends/onnxrt.py:8]: Without ort-dependencies installed, we throw detailed troubleshooting steps.
```
  File "/bert_ort/wechi/move-dort/pytorch/torch/_dynamo/backends/onnxrt.py", line 14, in information_displaying_backend
    raise ImportError(
torch._dynamo.exc.BackendCompilerFailed: backend='onnxrt' raised:
ImportError: onnxrt is not registered as a backend. Please make sure all dependencies such as numpy, onnx, onnxscript, and onnxruntime-training are installed. Suggested procedure to fix dependency problem: (1) pip or conda install numpy onnx onnxscript onnxruntime-training. (2) open a new python terminal (3) Run `from torch.onnx._internal.onnxruntime import has_onnxruntime`. (4) Run `has_onnxruntime()`. (5) If has_onnxruntime() returns True, then you can use `onnxrt` backend. (6) If has_onnxruntime() returns False, please execute the package importing section in torch/onnx/_internal/onnxruntime.py. under pdb line-by-line to see which import fails.
```

**Review by abock — APPROVED** [review]
❤️🎉

**Comment by thiagocrepaldi** [comment]
@malfet or @kit1980 might help too

**Review by msaroufim — APPROVED** [review]

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**